src/db.ts

```diff
@@ -724,15 +725,15 @@ export class Db {
 // Validate the database name
 function validateDatabaseName(databaseName: string) {
   if (typeof databaseName !== 'string')
-    throw new MongoDriverError('database name must be a string');
+    throw new MongoInvalidArgumentError('database name must be a string');
   if (databaseName.length === 0)
-    throw new MongoDriverError('database name cannot be the empty string');
+    throw new MongoInvalidArgumentError('database name cannot be the empty string');
   if (databaseName === '$external') return;
 
   const invalidChars = [' ', '.', '$', '/', '\\'];
   for (let i = 0; i < invalidChars.length; i++) {
     if (databaseName.indexOf(invalidChars[i]) !== -1)
-      throw new MongoDriverError(
+      throw new MongoInvalidArgumentError(
```

Comment:
This falls under the [earlier convo](https://github.com/mongodb/node-mongodb-native/pull/2891#issuecomment-876668400) we had. I wanted raise this so we can be consistent for whatever decision we make.